core/src/impl/Kokkos_StringManipulation.hpp

```diff
@@ -79,6 +79,7 @@ KOKKOS_INLINE_FUNCTION constexpr char *strncpy(char *dest, const char *src,
   if (count != 0) {
     char *d = dest;
     do {
+      // NOLINTNEXTLINE(bugprone-assignment-in-if-condition)
       if ((*d++ = *src++) == '\0') {
```

Comment:
We could also use the if init statement to avoid the warning:
```cpp
if (char const c = (*d++ = *src++); c == '\0')
```

Reply:
Sure